src/pool-cl/CLProtocolFeeController.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright (C) 2024 PancakeSwap
+pragma solidity 0.8.26;
+
+import {Ownable} from "../base/Ownable.sol";
+import {PoolKey} from "../types/PoolKey.sol";
+import {Currency} from "../types/Currency.sol";
+import {LPFeeLibrary} from "../libraries/LPFeeLibrary.sol";
+import {IProtocolFeeController} from "../interfaces/IProtocolFeeController.sol";
+import {IProtocolFees} from "../interfaces/IProtocolFees.sol";
+
+/// @notice ProtocolFeeController for CL Pool type
+contract CLProtocolFeeController is IProtocolFeeController, Ownable {
+    address public immutable clPoolManager;
+
+    event ProtocolFeeCollected(Currency indexed currency, uint256 amount);
+
+    error InvalidPoolManager();
+
+    constructor(address _clPoolManager) Ownable(msg.sender) {
+        clPoolManager = _clPoolManager;
+    }
+
+    /// @notice Get the protocol fee for a pool given the conditions of this contract
+    /// @return protocolFee The pool's protocol fee, expressed in hundredths of a bip. The upper 12 bits are for 1->0
+    function protocolFeeForPool(PoolKey memory poolKey) external pure override returns (uint24 protocolFee) {
```

Comment:
Do we need some kind of whitelist here ?  For example for some specific token pairs, we might want to manually give its initial protocolFee